Testing

far0uq opened the pull request.

far0uq committed:
update: refactoring#01

far0uq committed:
refactoring#02

far0uq committed:
update: responsive images added, close cart button and search query fix

Images once previously not shown for tablet or phone screens is now being shown for the both of them. Close Cart button not visible on phone made closing the cart impossible. Fixed it for smaller screens. And finally fixed a bug with the Search where when having no category, it wouldn't properly search the textFilter

far0uq committed:
test: configured testing and implemented tests for the Item Component using Mock Zustand Store

far0uq committed:
test: Wrote Unit tests for Item and CartContainer components

far0uq committed:
test: cartcontainer ~ added testing when the order is successfully calculated

far0uq committed:
update: improved calculation of order

Auto calculate order again every time the enabled discounts/taxes change. Also do this whenever the quantities are toggled.

Only have to calculate order manually the first time.

far0uq committed:
test: added the test to see if individual items prices are being calculated correctly

Check if Tax + Discounts + (Raw Value * Quantity) returns a true from the UI or not

mohammadahmadteq commented on __tests__/CartContainer.test.tsx:
```diff
@@ -0,0 +1,286 @@
+      container.getElementsByClassName(orderErrorText)[0] ?? null;
+    const orderLoadingElement =
+      container.getElementsByClassName(orderLoadingTest)[0] ?? null;
+    const orderEmptyElement =
```
Remove this check and add toBeFalsy excepts

mohammadahmadteq commented on src/app/clientAPI/productAPI.ts:
```diff
@@ -0,0 +1,46 @@
+const getFilteredProducts = async (
```
Can be refactored as a hook utilizing tanstack

mohammadahmadteq reviewed (changes requested):
Check the comments for changes regarding the nullish check in unit tests.

If possib;e refactor the api calls to use tanstack everywhere

far0uq committed:
update: created custom hook for item fetch, changed test and added error boundary

Custom Hook in place for Item Fetching. Changed the Tests to check if items are undefined instead of null. Added Sentry Error Boundary for error monitoring.

far0uq committed:
refactoring: moved all fetch tanstack queries and their relevant code to their own custom hooks

Removed the GET endpoint for Products/Items since it was useless. POST with empty body can perform the same functionality

far0uq committed:
update: add multiple custom hooks + big fixes

Created multiple custom hooks. useCartMutation, useFetchDiscounts, useFetchTaxes.

Had an issue where Order would not be calculated if a 'Free' element was in the cart. Turns out the Free element was actually 'Variable' price which means it would not have a fixed one. So we cant calculate for it. Handled this in the UI where Variable Cost items cannot be added to the Cart.

Changed the order calculation logic to auto calculate order and not on the basis of a button event.

far0uq committed:
update: tests fixed according to new changes

far0uq committed:
update: changed the import structure for matchMedia

far0uq committed:
update: add to cart button and place order

Add to Cart button added for initial cart addition. Toggler becomes active afterwards. Button in place in the cart to confirm and place the Order. Complete with Toast Message.

far0uq committed:
update: dual token & logout implemented

Now using two tokens, one for verifying jwt for middleware, one for using the API itself as square uses incorrect payload. So had to make a correct one as well for middleware verification. Logout also implemented where session is handled and destroyed.

far0uq committed:
update: fully functional auth pipeline

Authentication fixed and functional with proper logging in screen. Configured the AntD provider to work over forms. Need to make it so that the new page is not opened in a new window but in the same one

far0uq committed:
fix: routes successfully protected in middleware
